WordPress/Classes/ViewRelated/System/Action Sheet/DrawerPresentationController.swift

```diff
@@ -113,10 +132,36 @@ public class DrawerPresentationController: FancyAlertPresentationController {
 
         var frame = containerView.frame
         let y = collapsedYPosition
+        var width: CGFloat = containerView.bounds.width - (containerView.safeAreaInsets.left + containerView.safeAreaInsets.right)
 
         frame.origin.y = y
 
-        return frame
+        /// If we're in a compact vertical size class, constrain the width a bit more so it doesn't get overly wide.
+        if let widthForCompactSizeClass = presentableViewController?.width,
+            traitCollection.verticalSizeClass == .compact {
+
+            switch widthForCompactSizeClass {
+            case .percentage(let percentage):
+                width = width * percentage
+            case .contentWidth(let givenWidth):
+                width = givenWidth
+            case .maxWidth:
+                break
+            }
+        }
+
+        /// If we constrain the width, this centers the view by applying the appropriate insets based on width
+        let leftInset: CGFloat = ((containerView.bounds.width - width) / 2)
+
+        return CGRect(x: leftInset, y: frame.origin.y, width: width, height: frame.height)
```

Comment:
I don't think we need to create a new CGRect here. We can just modify the values of `frame` directly.

```
/// If we constrain the width, this centers the view by applying the appropriate insets based on width
frame.origin.x = ((containerView.bounds.width - width) / 2)
frame.size.width = width

return frame
```

Reply:
Done in f666e7c1888220008100034ac53c287bc19c598e